internal/db/user_reference.go

```diff
@@ -0,0 +1,95 @@
+package db
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/edulinq/autograder/internal/model"
+)
+
+// Process a list of user inputs in the context of a course.
+// See model.CourseUserReferenceInput for the list of acceptable inputs.
+// Inputs are parsed without checking permissions.
+// APIs using this function should require a core.MinCourseRoleGrader or custom permissions checking.
```

Comment:
Requiring `MinCourseRoleGrader` is not set in stone.
Keep the comment a little more general, like:
"API endpoints using this function should require a sufficient role or ...".